src/output-json-arp.c

```diff
@@ -0,0 +1,256 @@
+            return "mil_std_188_220";
+        case 23:
+            return "metricom";
+        case 24:
+            return "ieee1394.1995";
+        case 25:
+            return "mapos";
+        case 26:
+            return "twinaxial";
+        case 27:
+            return "eui64";
+        case 28:
+            return "hiparp";
+        case 29:
+            return "ip_over_arp";
+        case 30:
+            return "arpsec";
+        case 31:
+            return "ipsec";
+        case 32:
+            return "infiniband";
+        case 33:
+            return "cai";
+        case 34:
+            return "wiegand_interface";
+        case 35:
+            return "pure_ip";
+        case 36:
+            return "hw_exp1";
+        case 37:
+            return "hw_exp2";
+        case 38:
+            return "unified_bus";
+        case 256:
+            return "hw_exp2";
+        case 257:
+            return "aethernet";
+        default:
+            return "unknown";
+    }
+}
+
+static const char *prototype2str(uint16_t proto)
+{
+    switch (proto) {
+        case 0x0800:
+            return "ipv4";
+        default:
+            return "unknown";
+    }
+}
+
+static const char *opcode2str(uint16_t opcode)
+{
+    switch (opcode) {
+        case 0:
+        case 65535:
+            return "reserved";
+        case 1:
+            return "request";
+        case 2:
+            return "reply";
+        case 3:
+            return "request_reverse";
+        case 4:
+            return "reply_reverse";
+        case 5:
+            return "drarp_request";
+        case 6:
+            return "drarp_reply";
+        case 7:
+            return "drarp_error";
+        case 8:
+            return "inarp_request";
+        case 9:
+            return "inarp_reply";
+        case 10:
+            return "arp_nak";
+        case 11:
+            return "mars_request";
+        case 12:
+            return "mars_multi";
+        case 13:
+            return "mars_mserv";
+        case 14:
+            return "mars_join";
+        case 15:
+            return "mars_leave";
+        case 16:
+            return "mars_nak";
+        case 17:
+            return "mars_unserv";
+        case 18:
+            return "mars_sjoin";
+        case 19:
+            return "mars_sleave";
+        case 20:
+            return "mars_grouplist_request";
+        case 21:
+            return "mars_grouplist_reply";
+        case 22:
+            return "mars_redirect_map";
+        case 23:
+            return "mapos_unarp";
+        case 24:
+            return "op_exp1";
+        case 25:
+            return "op_exp2";
+        default:
+            return "unknown";
+    }
+}
+
+static int JsonArpLogger(ThreadVars *tv, void *thread_data, const Packet *p)
+{
+    OutputJsonThreadCtx *thread = thread_data;
+    char srcip[46] = "";
+    char dstip[46] = "";
+
+    JsonBuilder *jb = CreateEveHeader((Packet *)p, LOG_DIR_PACKET, "arp", NULL, thread->ctx);
```

Comment:
this cast is wrong, `CreateEveHeader` takes a const Packet pointer. `git grep` suggest we have more offending code.